syncthing/syncthing
# Don't create empty ".stignore" files (fixes #6190)

### Purpose

This will
1. Prevent creation of a new .stignore file if there are no ignore patterns (fixes #6190).
2. Delete an existing .stignore if all ignore patterns are removed.

### Testing

An automated test case is included with this pull request.  
If desired the changes can be tested manually as follows:
1. Go to the Syncthing GUI.
2. Edit a folder that doesn't have any ignore patterns.
3. Click "save" without making any changes. 
4. Observe that no .stignore file was created.
5. Edit the folder again and add an ignore pattern.
6. Observe that a .stignore file was created.
7. Edit the folder yet again and remove all ignore patterns.
8. Observe that the .stignore file was deleted.

### A note on software design
I'm undecided if the new `DeleteIgnores()` function in `lib/ignore/ignore.go` adds unnecessary complexity and whether it's function body should be inlined in `lib/model/model.go` instead.
Should you consider merging this PR I'd kindly request some feedback on this.
